edge_prediction_experiment/01_hetionet_recall_dag.py

```diff
@@ -0,0 +1,348 @@
+    test_entity_df >> ply.query("metric=='pred_median'") >> ply.pull("score"),
+)
+
+performance_map["PR"] = performance_map["PR"].append(
+    pd.DataFrame(
+        {
+            "precision": precision,
+            "recall": recall,
+            "pr_threshold": np.append(pr_threshold, 1),
+        }
+    )
+    >> ply.define(model=f'"pred_median/AUC={auc(recall, precision):.2f}"')
+)
+
+performance_map["AUROC"] = performance_map["AUROC"].append(
+    pd.DataFrame({"fpr": fpr, "tpr": tpr, "roc_threshold": roc_threshold})
+    >> ply.define(model=f'"pred_median/AUC={auc(fpr, tpr):.2f}"')
+)
+# -
+
+g = (
+    p9.ggplot(
+        performance_map["AUROC"]
+        >> ply.call(
+            ".append",
+            pd.DataFrame(
+                {
+                    "fpr": [0, 0.25, 0.5, 0.75, 1],
+                    "tpr": [0, 0.25, 0.5, 0.75, 1],
+                    "model": "random",
+                }
+            ),
+        )
+    )
+    + p9.aes(x="fpr", y="tpr", group="model", color="model", linetype="model")
+    + p9.geom_line()
+    + p9.theme_seaborn("white")
+    + p9.scale_color_manual(["#1b9e77", "#d95f02", "#7570b3", "#000000"])
+    + p9.scale_linetype_manual(["solid", "solid", "solid", "dashed"])
+)
+print(g)
+
+g = (
+    p9.ggplot(performance_map["PR"])
+    + p9.aes(x="recall", y="precision", group="model", color="model", linetype="model")
+    + p9.geom_line()
+    + p9.theme_seaborn("white")
+    + p9.scale_color_manual(["#1b9e77", "#d95f02", "#7570b3", "#000000"])
+    + p9.scale_linetype_manual(["solid", "solid", "solid", "dashed"])
+    + p9.geom_hline(
+        yintercept=(
+            test_entity_df >> ply.query("metric=='pred_median'") >> ply.pull("hetionet")
+        ).mean(),
+        linetype="dashed",
+    )
+)
+print(g)
+
+# ## Estimate number of new Edges Added
+
+# +
+df_iterator = (
+    performance_map["PR"]
+    >> ply.query("model.str.contains('max')& pr_threshold < 1")
+    >> ply.call(".round", {"pr_threshold": 2})
+    >> ply.distinct("pr_threshold", "last")
+)
+
+edges_added_records = []
+for idx, row in df_iterator.iterrows():
+    cutoff = row["pr_threshold"]
+
+    values_added = (
+        all_dag_df
+        >> ply.query("metric.str.contains('max')")
+        >> ply.query("score >= @cutoff")
+        >> ply.pull("hetionet")
+    )
+
+    edges_added_records.append(
+        {
+            "edges": values_added.sum(),
+            "in_hetionet": "Existing",
+            "precision": row["precision"],
+            "sen_cutoff": cutoff,
+        }
+    )
+
+    edges_added_records.append(
+        {
+            "edges": values_added.shape[0] - values_added.sum(),
+            "in_hetionet": "Novel",
+            "precision": row["precision"],
+            "sen_cutoff": cutoff,
+        }
+    )
+
+
+edges_added_df = pd.DataFrame.from_records(edges_added_records)
+edges_added_df >> ply.slice_rows(10)
+# -
+
+g = (
+    p9.ggplot(edges_added_df >> ply.query("edges > 0 & precision > 0"))
```

Comment:
In this plot, is it that the more stringent you make your threshold for deciding where an edge exists, the higher the precision? And with the large drop off in new edges, that means that its predicting many FP?

Reply:
>In this plot, is it that the more stringent you make your threshold for deciding where an edge exists, the higher the precision? And with the large drop off in new edges, that means that its predicting many FP?

Yup. That's the trade off for precision. The more stringent your threshold the higher the precision, but lower the recall. FP in this case may not be true FP as it could be new edge types; however, for the quick analysis this is the best way to figure threshold cutoffs.